### PR TITLE
Menu item to view a community in a specific instance other than the originating instance

### PR DESCRIPTION
**Pull request by jcgurango**
This adds a new menu item which appears under the "..." menu of a community called "View on another instance".
image

This pops up a list of accounts for you to choose from. The "Subscribe" button, when viewing anonymously, also pops this up:
image

This then executes a ResolveObject call to retrieve the community:
image

Before showing you the community, in the other instance. Denoted by the "(via instance)" text in the name of the community.
image

You can then subscribe directly from here, or if some posts are loaded you can even comment or vote directly.

Dependent on #261

### CODE REVIEW

File: lib/pages/community/foreign_community.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:lemmy_api_client/v3.dart';
+
+import '../../l10n/l10n.dart';
+import '../../stores/accounts_store.dart';
+import 'community.dart';
+
+class ForeignCommunityPage extends HookWidget {
```
Comment (zachatrocity) [inline lib/pages/community/foreign_community.dart:9]: Could we name this like.. FederatedCommunityPage? I'd rather not introduce a new concept

**Review by zachatrocity — COMMENTED** [review]
This looks really really good. Nice work on a big challenge. 

Most of the changes are around updating jwt to user data which I love cause it gives us a lot of room to scale.

Sorry to be this guy but given that this is such a critical piece of our app, we need to add some unit tests that test the useLoggedInAction functionality and also your new subscribe capability

**Comment by jcgurango** [comment]
@zachatrocity There's actually no additional functionality added to the subscribe button other than opening the menu for non-logged in instances. I also don't think there should be because it would abstract away the fact that it's not on their instance. Noted on the useLoggedInAction hook

**Comment by zachatrocity** [comment]
I see, really usedLoggedInAction should have had a test before. 

I'll approve, having a test would be nice but we don't have much scaffolding out there.

**Review by zachatrocity — APPROVED** [review]
LGTM, just need to fix the lint

**Pull request merged**